In [1]:
import pandas as pd
import numpy as np


In [31]:
data=pd.read_csv("train.csv")
data.head()

Deal_title                      Lead_name               Industry  \
0  TitleM5DZY  Davis, Perkins and Bishop Inc            Restaurants   
1  TitleKIW18                 Bender PLC LLC  Construction Services   
2  TitleFXSDN          Carter-Henry and Sons      Hospitals/Clinics   
3  TitlePSK4Y                 Garcia Ltd Ltd            Real Estate   
4  Title904GV               Lee and Sons PLC     Financial Services   

  Deal_value Weighted_amount Date_of_creation      Pitch  \
0    320506$      2067263.7$       2020-03-29  Product_2   
1     39488$       240876.8$       2019-07-10  Product_2   
2    359392$      2407926.4$       2019-07-27  Product_1   
3     76774$       468321.4$       2021-01-30  Product_2   
4    483896$             NaN       2019-05-22  Product_2   

             Contact_no             Lead_revenue Fund_category  ...  \
0          607.447.7883         50 - 100 Million    Category 2  ...   
1          892-938-9493  500 Million - 1 Billion    Category 4  ...   
2          538.748.2271  500 Million - 1 Billion    Category 4  ...   
3   (692)052-1389x75188  500 Million - 1 Billion    Category 3  ...   
4  001-878-814-6134x015         50 - 100 Million    Category 3  ...   

                Designation                 Lead_POC_email  \
0  Executive Vice President       charlenewerner@davis.com   
1    Chairman/CEO/President          terrylogan@bender.com   
2      SVP/General Counsel   arielhamilton@carterhenry.com   
3   CEO/Co-Founder/Chairman          erinwilson@garcia.com   
4  Executive Vice President         mr.christopher@lee.com   

            Hiring_candidate_role      Lead_source Level_of_meeting  \
0            Community pharmacist          Website          Level 3   
1          Recruitment consultant           Others          Level 1   
2          Health service manager  Marketing Event          Level 1   
3  Therapist, speech and language    Contact Email          Level 2   
4                   Media planner          Website          Level 2   

                  Last_lead_update               Internal_POC  \
0                         No track           Davis,Sharrice A   
1  Did not hear back after Level 1             Brown,Maxine A   
2                                ?  Georgakopoulos,Vasilios T   
3  Did not hear back after Level 1             Brown,Maxine A   
4                       Up-to-date              Thomas,Lori E   

                       Resource Internal_rating Success_probability  
0                           NaN               3                73.6  
1                            No               5                58.9  
2                            No               4                68.8  
3  We have all the requirements               1                64.5  
4                            No               4                62.4  

[5 rows x 23 columns]

In [32]:
print("Shape of our data set ",data.shape)

Shape of our data set  (7007, 23)


In [33]:
#check data type of all the coulmns
data.dtypes

Deal_title                object
Lead_name                 object
Industry                  object
Deal_value                object
Weighted_amount           object
Date_of_creation          object
Pitch                     object
Contact_no                object
Lead_revenue              object
Fund_category             object
Geography                 object
Location                  object
POC_name                  object
Designation               object
Lead_POC_email            object
Hiring_candidate_role     object
Lead_source               object
Level_of_meeting          object
Last_lead_update          object
Internal_POC              object
Resource                  object
Internal_rating            int64
Success_probability      float64
dtype: object

In [34]:
#check for null values
data.isnull().sum()

Deal_title                 0
Lead_name                  0
Industry                   1
Deal_value                51
Weighted_amount          525
Date_of_creation           0
Pitch                      0
Contact_no                 0
Lead_revenue               0
Fund_category              0
Geography                972
Location                  11
POC_name                   8
Designation                0
Lead_POC_email             0
Hiring_candidate_role      0
Lead_source                0
Level_of_meeting           0
Last_lead_update         633
Internal_POC               0
Resource                 149
Internal_rating            0
Success_probability        0
dtype: int64

# First we will treat all missing values

In [35]:
#check null value 4653 Title232C0
data[data["Industry"].isnull()]

Deal_title                                  Lead_name Industry  \
4653  Title232C0  Valenzuela, Williamson and Henry and Sons      NaN   

     Deal_value Weighted_amount Date_of_creation      Pitch    Contact_no  \
4653    209418$      1266978.9$       2020-12-15  Product_2  737.346.8367   

          Lead_revenue Fund_category  ...           Designation  \
4653  50 - 100 Million    Category 4  ...  SVP/General Counsel    

                  Lead_POC_email Hiring_candidate_role    Lead_source  \
4653  johnathanle@valenzuela.com   Landscape architect  Contact Email   

     Level_of_meeting Last_lead_update    Internal_POC Resource  \
4653          Level 1      2 days back  Hameier,Kurt E       No   

     Internal_rating Success_probability  
4653               3                56.6  

[1 rows x 23 columns]

In [36]:
# task 1 Fill missing value in Industry
data["Industry"] = data["Industry"].fillna((data["Industry"].mode()[0]))

In [37]:
# Deal_value consist of 51 missing values we will impute mean over missing values.
data[data["Deal_value"].isnull()]

Deal_title                                 Lead_name  \
169   TitleKXH1L               Turner, Green and Brown Ltd   
381   TitleBV4VN                 Perez, Lopez and Lara Inc   
448   TitleMUK48            Harding, Burke and Price Group   
469   TitleXSNO6                      Marshall-Johnson LLC   
519   TitleK69MY                     Campbell Ltd and Sons   
1016  TitleXD5MP                  Burke, Hart and Hale PLC   
1148  TitleWB44B                    Ruiz-Mcdaniel and Sons   
1246  Title6X0QM           Alvarez, Cooper and Griffin Inc   
1328  TitleDUHSS                      Ramirez-Figueroa LLC   
1587  TitleMPX29                  Walker-Browning and Sons   
1599  Title27F7I                              Duke Ltd LLC   
1626  TitleO80HC                        Smith Ltd and Sons   
1636  TitleNSRAK                  Wood, Smith and Rice LLC   
1651  TitleSGTCT           Mccormick, Norris and Clark Ltd   
1849  Title8XSJJ        Matthews, Fields and Barnett Group   
1943  Title62ICV                       Chambers-Hurley Ltd   
2039  TitleXQ4QR                           Schultz LLC Inc   
2165  TitlePC79J                    Thompson-Ward and Sons   
2241  TitleIUOP5                          Hobbs-Kelley PLC   
2469  TitleTYPR7        Mcconnell, Pacheco and Moyer Group   
2487  TitleDGY3M                        Ramos-Knight Group   
2869  TitleZZXEA                       Rivera-Greene Group   
2953  TitleFF3TA        Gilbert, Russell and Hale and Sons   
3139  TitleGUEIA                         Robertson PLC PLC   
3272  TitleAXESK                           Wu and Sons Inc   
3327  TitleNMDJA                             Lopez Inc PLC   
3403  TitleZELYQ                           Martinez-Yu PLC   
3586  Title83O8F              Armstrong-Mcfarland and Sons   
3632  TitleZFOVB                           James Inc Group   
3919  TitleRWOK0             Mcknight, Wise and Brooks Inc   
4269  Title8T56L                         Johnson-Hayes LLC   
4300  TitleQLI4G               Solomon, Khan and Baker LLC   
4722  TitleFWISC                           Ramirez Inc Ltd   
4745  TitleMLDBZ                 Brown-Washington and Sons   
4926  Title371LP                             Lewis PLC Inc   
5276  TitleVMPZG             Garrett, Gates and Newman Ltd   
5401  Title1DHCN                         Flores-Snyder PLC   
5451  Title49N0C                            Heath-Lane PLC   
5620  TitleL45YI             Villanueva, Diaz and Shea Ltd   
5889  TitleCA8OV                           Simmons LLC PLC   
6001  TitleJ28M7                       Chavez-Humphrey LLC   
6100  TitleEUOPL            Miller, Smith and Murphy Group   
6115  TitleZCNWC          Moran, Shaw and Johnson and Sons   
6213  Title43EZV                          Cole-Burnett Inc   
6423  TitleNBT4E      Torres, Lewis and Gutierrez and Sons   
6501  Title445PS         Jefferson, Gardner and Byrd Group   
6567  Title1P6RF                      Dickerson-Garner Inc   
6611  TitleRFVSJ                        Patrick-Moreno Inc   
6653  TitleODFVR                         Wilson-Bailey Inc   
6878  TitleJ8EL1       Alvarez, Anderson and Sanders Group   
6885  Title8JHPF  Preston, Sanchez and Mccullough and Sons   

                            Industry Deal_value Weighted_amount  \
169                            Banks        NaN      1023175.8$   
381                       Iron/Steel        NaN       668538.0$   
448                      Real Estate        NaN             NaN   
469                         Software        NaN     1167666.85$   
519                            Banks        NaN      2839513.6$   
1016          Network Infrastructure        NaN      2929018.9$   
1148              Furniture/Fixtures        NaN     2432947.95$   
1246                        Staffing        NaN      1195450.0$   
1328                       Insurance        NaN     2607096.45$   
1587                           Music        NaN      408901.35$   
1599              Education/Training        

To fill missing value in Deal_Value column we first need to conver it to integer. For that we have to remove that $ sign first. 

In [39]:
data["Deal_value"]= data["Deal_value"].str.replace("$","")
data.Deal_value.head()

<ipython-input-39-e80ee3952f61>:1: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will*not* be treated as literal strings when regex=True.
  data["Deal_value"]= data["Deal_value"].str.replace("$","")


0    320506
1     39488
2    359392
3     76774
4    483896
Name: Deal_value, dtype: object

In [42]:
#converting Deal_Value to numeric
data["Deal_value"] = pd.to_numeric(data["Deal_value"])

0    320506.0
1     39488.0
2    359392.0
3     76774.0
4    483896.0
Name: Deal_value, dtype: float64

In [43]:
#imputing missing value in Deal_value
data["Deal_value"] = data["Deal_value"].fillna(data["Deal_value"].mean())

In [46]:
#Similarly for Weighted Amount we have to remove the $ sign first then impute the missing values with mean 
data[data["Weighted_amount"].isnull()]

Deal_title                        Lead_name                   Industry  \
4     Title904GV                 Lee and Sons PLC         Financial Services   
13    TitleNQKU6           Fowler-Medina and Sons                      Banks   
17    TitleEWIQD                Rodriguez Inc LLC                   Software   
23    TitleO0W9A                   Meza-Myers Inc                       REIT   
33    TitleQ5K7M   Petersen, Lindsey and Wong LLC  Automotive/Transportation   
...          ...                              ...                        ...   
6971  Title6SSU1   Cortez, Manning and Morris LLC          Hospitals/Clinics   
6975  Title5441H                Whitehead LLC PLC         Financial Services   
6983  TitleVDKHN                  Henson-Soto Ltd      Professional Services   
6990  Title4H88W  Miranda, Mccoy and Mccarthy Inc                  Insurance   
7004  TitleF1FWY                Romero-Juarez PLC             Semiconductors   

      Deal_value Weighted_amount Date_of_creation      Pitch  \
4       483896.0             NaN       2019-05-22  Product_2   
13      487351.0             NaN       2020-04-25  Product_1   
17      140949.0             NaN       2019-05-19  Product_1   
23      207335.0             NaN       2020-10-27  Product_1   
33      335694.0             NaN       2019-10-04  Product_2   
...          ...             ...              ...        ...   
6971    101621.0             NaN       2019-05-14  Product_1   
6975    456307.0             NaN       2020-04-03  Product_2   
6983     89949.0             NaN       2019-07-07  Product_1   
6990    240326.0             NaN       2020-11-23  Product_2   
7004    253608.0             NaN       2020-03-10  Product_1   

                 Contact_no             Lead_revenue Fund_category  ...  \
4      001-878-814-6134x015         50 - 100 Million    Category 3  ...   
13    001-149-287-9330x6974  500 Million - 1 Billion    Category 3  ...   
17             667-032-4850        100 - 500 Million    Category 3  ...   
23      +1-025-771-0426x190        100 - 500 Million    Category 3  ...   
33             616-555-1222         50 - 100 Million    Category 3  ...   
...                     ...                      ...           ...  ...   
6971      (271)826-3076x520         50 - 100 Million    Category 1  ...   
6975       067.825.7189x868  500 Million - 1 Billion    Category 2  ...   
6983      553-512-4648x4881        100 - 500 Million    Category 2  ...   
6990     269-108-7756x15312        100 - 500 Million    Category 4  ...   
7004       777-231-4109x712        100 - 500 Million    Category 2  ...   

                            Designation                   Lead_POC_email  \
4              Executive Vice President           mr.christopher@lee.com   
13    Chairman/Chief Innovation Officer   raymondpeters@fowlermedina.com   
17             Executive Vice President  christopherwalter@rodriguez.com   
23               CEO/Chairman/President         dannyadams@mezamyers.com   
33                        CEO/President              alexle@petersen.com   
...                                 ...                              ...   
6971                                CEO          jeffreysmith@cortez.com   
6975             CEO/Chairman/President        michaelbuck@whitehead.com   
6983               SVP/General Counsel       marycummings@hensonsoto.com   
6990           Executive Vice President          josephallen@miranda.com   
7004               SVP/General Counsel   nicholasduncan@romerojuarez.com   

              Hiring_candidate_role      Lead_source Level_of_meeting  \
4                     Media planner          Website          Level 2   
13                  Hospital doctor  Marketing Event          Level 3   
17                   Phytotherapist  Marketing Event          Level 2   
23                    Warden/ranger          Website          Level 2   
33    Operational investment banker    Contact Email          Level 1   
...                     

In [48]:
# removing the $ sign
data["Weighted_amount"]= data["Weighted_amount"].str.replace("$","")

<ipython-input-48-826d9348be3e>:2: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will*not* be treated as literal strings when regex=True.
  data["Weighted_amount"]= data["Weighted_amount"].str.replace("$","")


In [66]:
# now conver the Weighted_amount to numeric and then impute missing values using mean()
#converting Weighted_amount to numeric
data["Weighted_amount"] = pd.to_numeric(data["Weighted_amount"])
data["Weighted_amount"] = data["Weighted_amount"].fillna(data["Weighted_amount"].mean())

In [79]:
#data.head()
#data[data["Deal_title"]=="TitleKXH1L"]

In [75]:
#converting the Date_of_creation to date time
data["Date_of_creation"] = pd.to_datetime(data["Date_of_creation"])
data.Date_of_creation.dtype

dtype('<M8[ns]')

In [95]:
#to get all columns
pd.set_option("display.max_columns",25)

In [103]:
data[data["Geography"].isnull()].head()

Deal_title                          Lead_name  \
6   TitleOZQRY            Williamson LLC and Sons   
10  TitleP5EAQ               Bradley-Willis Group   
13  TitleNQKU6             Fowler-Medina and Sons   
16  TitleBZ8I3                     Hull Group Ltd   
27  TitleWKMYP  Maldonado, Doyle and Patton Group   

                          Industry  Deal_value  Weighted_amount  \
6                            Banks    384356.0     2.709710e+06   
10  Healthcare Facilities/Services    311125.0     1.851194e+06   
13                           Banks    487351.0     1.569884e+06   
16          Other Investment Firms    390309.0     2.458947e+06   
27                        Trucking    380817.0     2.513392e+06   

   Date_of_creation      Pitch             Contact_no  \
6        2019-11-20  Product_2          (881)077-4692   
10       2020-01-30  Product_1      466.038.3610x6981   
13       2020-04-25  Product_1  001-149-287-9330x6974   
16       2019-10-13  Product_2   +1-369-535-1832x6894   
27       2019-05-31  Product_2           650.070.5743   

               Lead_revenue Fund_category Geography  \
6   500 Million - 1 Billion    Category 1       NaN   
10  500 Million - 1 Billion    Category 2       NaN   
13  500 Million - 1 Billion    Category 3       NaN   
16         50 - 100 Million    Category 2       NaN   
27         50 - 100 Million    Category 1       NaN   

                                     Location        POC_name  \
6                            Salisbury, MD-DE      Sara Dixon   
10           Oxnard-Thousand Oaks-Ventura, CA  Tamara Sanchez   
13                                   Pilibhit         saaniya   
16  Miami-Fort Lauderdale-West Palm Beach, FL   Joshua Wright   
27                Chambersburg-Waynesboro, PA  Jennifer Davis   

                          Designation                   Lead_POC_email  \
6             CEO/Co-Founder/Chairman         saradixon@williamson.com   
10            CEO/Co-Founder/Chairman  tamarasanchez@bradleywillis.com   
13  Chairman/Chief Innovation Officer   raymondpeters@fowlermedina.com   
16             Chairman/CEO/President            joshuawright@hull.com   
27                      CEO/President      jenniferdavis@maldonado.com   

               Hiring_candidate_role      Lead_source Level_of_meeting  \
6                       Cartographer    Contact Email          Level 3   
10                    Hydrogeologist          Website          Level 1   
13                   Hospital doctor  Marketing Event          Level 3   
16              Broadcast journalist    Contact Email          Level 3   
27  Armed forces operational officer    Contact Email          Level 3   

                        Last_lead_update      Internal_POC  \
6                      More than 2 weeks    Booker,David L   
10  Following up but lead not responding  Clavey,Therese A   
13                 More than a week back  Shelton,Sidney P   
16                                     ?  Dimattia,Frank D   
27       Did not hear back after Level 1      Leu,Darren L   

                        Resource  Internal_rating  Success_probability  
6                            NaN                4                73.40  
10                Cannot deliver                2                67.90  
13  We have all the requirements                2                26.35  
16                   Deliverable                4                70.00  
27                            No                1                68.30

In [107]:
#we will replace the geopgraphy value by mode 
data["Geography"] = data["Geography"].fillna((data["Geography"].mode()[0]))

In [112]:
data[data["Location"].isnull()]

Deal_title                         Lead_name                   Industry  \
545   TitleY3XZP                 Ramirez-Grant Inc  Investment Bank/Brokerage   
810   TitleE0MD4                  Anderson Ltd PLC                       REIT   
1597  TitleX2NTQ   Flores, Rodriguez and Rubio LLC                  Insurance   
1623  Title4AI1N    Bryant, Powell and Benitez Inc                      Banks   
1698  TitleT9KKE                     Lopez LLC Inc         Telecom Consulting   
1901  TitleQF35M                  Baxter-Kelly Inc  Investment Bank/Brokerage   
3466  TitleHI98X       Bowen, Nolan and Thomas Inc   Architecture/Engineering   
4201  TitleK42F3  Price, Anderson and Robinson PLC         Other Biz Services   
5222  TitleOSYEE                  Wilson-Young LLC                   Staffing   
5675  Title9V907                   Vaughan Ltd PLC         Financial Services   
6862  Title70QKM                Stout and Sons Ltd                      Banks   

      Deal_value  Weighted_amount Date_of_creation      Pitch  \
545     321315.0       2056416.00       2019-06-30  Product_1   
810     457184.0       3108851.20       2019-03-30  Product_1   
1597    134465.0        793343.50       2020-10-07  Product_1   
1623    380753.0       2684308.65       2020-08-18  Product_1   
1698    383929.0       2687503.00       2020-01-14  Product_2   
1901     37105.0        222630.00       2019-09-27  Product_2   
3466    485381.0       3252052.70       2021-01-18  Product_2   
4201     15638.0        103992.70       2020-05-21  Product_2   
5222    449933.0       2924564.50       2020-12-04  Product_2   
5675    164299.0       1026868.75       2021-01-17  Product_1   
6862     63427.0        396418.75       2019-01-12  Product_2   

                  Contact_no             Lead_revenue Fund_category Geography  \
545        665.653.0803x9563        100 - 500 Million    Category 1       USA   
810             900-106-9151         50 - 100 Million    Category 1       USA   
1597        748-346-6890x165  500 Million - 1 Billion    Category 3       USA   
1623        663.813.6980x393  500 Million - 1 Billion    Category 3       USA   
1698  001-502-795-0451x38729         50 - 100 Million    Category 4       USA   
1901       457-737-0811x5558  500 Million - 1 Billion    Category 2       USA   
3466            240.497.2718  500 Million - 1 Billion    Category 4       USA   
4201      (944)525-1243x7415         50 - 100 Million    Category 4       USA   
5222     (793)446-2769x10877  500 Million - 1 Billion    Category 4       USA   
5675            186-435-4466        100 - 500 Million    Category 2       USA   
6862       (918)317-5630x163        100 - 500 Million    Category 3       USA   

     Location          POC_name  \
545       NaN   Jennifer Thomas   
810       NaN      Bianca Kelly   
1597      NaN     Colin Camacho   
1623      NaN    David Williams   
1698      NaN      Brendan Reid   
1901      NaN   Virginia Medina   
3466      NaN       Sarah Lloyd   
4201      NaN      Joseph Brown   
5222      NaN   Daniel Delacruz   
5675      NaN  Charles Robinson   
6862      NaN   Elizabeth House   

                                            Designation  \
545                                SVP/General Counsel    
810                             CEO/Co-Founder/Chairman   
1597                             Chairman/CEO/President   
1623                             CEO/Chairman/President   
1698  Vice President / GM (04-present) : VP Sales an...   
1901                             Chairman/CEO/President   
3466                             Chairman/CEO/President   
4201                  Chairman/Chief Innovation Officer   
5222                            CEO/Co-Founder/Chairman   
5675                           Executive Vice President   
6862                             CEO/Chairman/President   

                       Lead_POC_email            Hiring_candidate_role  \
545   jenniferthomas@ramirezgrant.com       Nurse, learning disability   
8

In [128]:
# The Location feature consist of around 11 missing values with region as USA and no missing values in the India region.
#Lets check the most common location is the USA region. 
data2=data[data["Geography"]=="USA"]
data2.Location.mode()

0    Panama City, FL
dtype: object

In [119]:
#now replace the missing value in the Location column where region is USA with Panama City, FL

In [137]:
data["Location"] = data["Location"].fillna((data2.Location.mode()[0]))

In [140]:
#data[data["Deal_title"]=="TitleY3XZP"]
#cross check the values

In [144]:
# now we have missing values in the POC_name column, Replace it with mode
data.POC_name.mode()[0]

'pooja'

In [148]:
data[data["POC_name"].isnull()].head()

Deal_title                          Lead_name         Industry  \
489   TitleS2D23  Martinez, Gutierrez and Knapp PLC        Insurance   
599   TitleST3S0      Anderson, Buckley and Lee LLC        Insurance   
1862  TitleDQJEL                  Armstrong Inc LLC            Banks   
2008  Title612IP      King, Davenport and James LLC  Servers/Storage   
2053  TitleEPQ2Q                   Lozano-Clark Inc            Banks   

      Deal_value  Weighted_amount Date_of_creation      Pitch  \
489     209984.0       1186409.60       2019-06-07  Product_2   
599     326899.0       2043118.75       2019-07-04  Product_1   
1862    457421.0       2813139.15       2019-06-04  Product_2   
2008    118302.0        727557.30       2020-03-10  Product_2   
2053     90309.0        546369.45       2021-02-07  Product_2   

               Contact_no             Lead_revenue Fund_category Geography  \
489          358-063-2204         50 - 100 Million    Category 3     India   
599            9576256452         50 - 100 Million    Category 1     India   
1862  +1-816-342-8073x681  500 Million - 1 Billion    Category 2     India   
2008           6521077903  500 Million - 1 Billion    Category 2     India   
2053   (032)148-1914x4137         50 - 100 Million    Category 4     India   

        Location POC_name               Designation  \
489   Alipurduar      NaN  Executive Vice President   
599        Sirsa      NaN   Chief Executive Officer   
1862    Raurkela      NaN      SVP/General Counsel    
2008       Daman      NaN   Chief Executive Officer   
2053   Thanjavur      NaN   CEO/Co-Founder/Chairman   

                    Lead_POC_email          Hiring_candidate_role  \
489    chelseajohnson@martinez.com  Interior and spatial designer   
599       williamcole@anderson.com      Teacher, secondary school   
1862   jamesmarshall@armstrong.com                   Statistician   
2008         wendygilbert@king.com           Commissioning editor   
2053  lisacaldwell@lozanoclark.com            Early years teacher   

          Lead_source Level_of_meeting   Last_lead_update      Internal_POC  \
489     Contact Email          Level 2        2 days back    Booker,David L   
599     Contact Email          Level 2  more than a month  Davis,Sharrice A   
1862    Contact Email          Level 3           No track  Dimattia,Frank D   
2008    Contact Email          Level 3            Pending     Gilley,Janine   
2053  Marketing Event          Level 1        5 days back       Ross,Eric L   

                          Resource  Internal_rating  Success_probability  
489                     Not enough                2                55.60  
599                    Deliverable                3                61.80  
1862                    Not enough                1                61.70  
2008  We have all the requirements                1                26.35  
2053                   Deliverable                1                56.70

In [149]:
#replace  missing value in POC_name
data["POC_name"] = data["POC_name"].fillna((data.POC_name.mode()[0]))

In [158]:
#filling 633 missing values in Last_lead_update with mode value
data.Last_lead_update.mode()[0]

'Following up but lead not responding'

In [160]:
data[data["Last_lead_update"].isnull()].head()

Deal_title                           Lead_name            Industry  \
12  TitleOGCTY                    Weaver Group Inc  Biotech/Healthcare   
35  TitleLLTG0                  Hughes-Johnson PLC   Constr - Supplies   
48  TitleZITZC                   Bass and Sons Inc               Banks   
51  Title190KW  Norton, Edwards and Chung and Sons        Associations   
65  Title8W9AG                  Howe-Cruz and Sons           Insurance   

    Deal_value  Weighted_amount Date_of_creation      Pitch  \
12    222331.0        1400685.3       2020-03-27  Product_2   
35    216892.0        1344730.4       2019-07-30  Product_1   
48    376485.0        2560098.0       2020-05-02  Product_2   
51    457274.0        2972281.0       2019-10-25  Product_2   
65     58778.0         405568.2       2020-02-17  Product_2   

              Contact_no             Lead_revenue Fund_category Geography  \
12          262-729-7106        100 - 500 Million    Category 2       USA   
35         (456)789-9864        100 - 500 Million    Category 3       USA   
48          554-929-7971         50 - 100 Million    Category 1       USA   
51  001-126-664-5288x091  500 Million - 1 Billion    Category 4     India   
65  +1-035-607-7888x3384  500 Million - 1 Billion    Category 4     India   

                            Location           POC_name  \
12                      Danville, IL          Frank Ali   
35                        Monroe, MI  Anthony Armstrong   
48  Kingsport-Bristol-Bristol, TN-VA         Amy Stuart   
51                        Saharanpur       puja chauhan   
65                 Tiruchchirappalli             kushum   

                                          Designation  \
12  Vice President / GM (04-present) : VP Sales an...   
35                             CEO/Chairman/President   
48  Vice President / GM (04-present) : VP Sales an...   
51                           Executive Vice President   
65                             CEO/Chairman/President   

                        Lead_POC_email          Hiring_candidate_role  \
12                 frankali@weaver.com  Speech and language therapist   
35  anthonyarmstrong@hughesjohnson.com        Housing manager/officer   
48                  amystuart@bass.com       Secondary school teacher   
51           jenniferguzman@norton.com     Psychologist, occupational   
65            ashleydixon@howecruz.com          Accountant, chartered   

        Lead_source Level_of_meeting Last_lead_update         Internal_POC  \
12  Marketing Event          Level 1              NaN         Maine,John P   
35           Others          Level 2              NaN        Davis,Brian R   
48    Contact Email          Level 3              NaN  McKenstry,Loretta A   
51  Marketing Event          Level 2              NaN       Smith,Keenan H   
65          Website          Level 3              NaN       Knox,Antonio D   

          Resource  Internal_rating  Success_probability  
12  Cannot deliver                5                71.50  
35  Cannot deliver                5                69.70  
48              No                3               107.34  
51             Yes                2                60.30  
65             Yes                4               107.34

In [161]:
data.Last_lead_update.mode()[0]

'Following up but lead not responding'

In [162]:
#replace  missing value in Last_lead_update
data["Last_lead_update"] = data["Last_lead_update"].fillna((data.Last_lead_update.mode()[0]))

In [164]:
#check missing values in Resource column
data.Resource.isnull().sum()

149

In [165]:
data[data["Resource"].isnull()].head()

Deal_title                      Lead_name  \
0    TitleM5DZY  Davis, Perkins and Bishop Inc   
6    TitleOZQRY        Williamson LLC and Sons   
111  TitleNP5Z6       Salinas-Parsons and Sons   
145  Title6BJEK                Foley-Berry PLC   
189  Title3ORBL                Ramirez LLC Ltd   

                           Industry  Deal_value  Weighted_amount  \
0                       Restaurants    320506.0     2.067264e+06   
6                             Banks    384356.0     2.709710e+06   
111                   Hotels/Motels    435551.0     2.417308e+06   
145  Healthcare Facilities/Services    283482.0     1.658370e+06   
189              Other Biz Services    321223.0     1.569884e+06   

    Date_of_creation      Pitch              Contact_no  \
0         2020-03-29  Product_2            607.447.7883   
6         2019-11-20  Product_2           (881)077-4692   
111       2020-06-02  Product_1            264-257-8983   
145       2020-09-24  Product_2  001-001-743-9237x61783   
189       2020-05-02  Product_2       262.927.0047x8177   

                Lead_revenue Fund_category Geography            Location  \
0           50 - 100 Million    Category 2       USA  Killeen-Temple, TX   
6    500 Million - 1 Billion    Category 1       USA    Salisbury, MD-DE   
111         50 - 100 Million    Category 2     India        Vizianagaram   
145         50 - 100 Million    Category 4     India            Adilabad   
189        100 - 500 Million    Category 1       USA           Chico, CA   

              POC_name               Designation  \
0      Charlene Werner  Executive Vice President   
6           Sara Dixon   CEO/Co-Founder/Chairman   
111               hina    Chairman/CEO/President   
145          asha devi                       CEO   
189  Darrell Torres MD    Chairman/CEO/President   

                        Lead_POC_email              Hiring_candidate_role  \
0             charlenewerner@davis.com               Community pharmacist   
6             saradixon@williamson.com                       Cartographer   
111  rebeccajohnson@salinasparsons.com  Special educational needs teacher   
145      michelemanning@foleyberry.com             Engineer, agricultural   
189          darrelltorres@ramirez.com        Careers information officer   

       Lead_source Level_of_meeting                      Last_lead_update  \
0          Website          Level 3                              No track   
6    Contact Email          Level 3                     More than 2 weeks   
111  Contact Email          Level 1  Following up but lead not responding   
145         Others          Level 1                               Pending   
189        Website          Level 2  Following up but lead not responding   

            Internal_POC Resource  Internal_rating  Success_probability  
0       Davis,Sharrice A      NaN                3                73.60  
6         Booker,David L      NaN                4                73.40  
111       Mabrey,Kevin C      NaN                3                58.60  
145        Charles,Caleb      NaN                4               107.34  
189  McKenstry,Loretta A      NaN                5                75.00

In [167]:
#most occuring value of Resource Category
data["Resource"].mode()[0]

'No'

In [168]:
#replace  missing value in Resource
data["Resource"] = data["Resource"].fillna((data["Resource"].mode()[0]))

# # Now all the missing values treatment is over. Next we have to encode the categorical column.

In [ ]:
# Save the file as processed File version 1 for a safer side.

In [174]:
data.to_csv("Processed_V1.csv", index=False)

In [198]:
#data[data["Success_probability"]>100].count()

# Starting with encoding the categorical values.

In [175]:
from sklearn.preprocessing import LabelEncoder
lab_encod= LabelEncoder()

In [176]:
#lets start with encoding.
data.head()

Deal_title                      Lead_name               Industry  \
0  TitleM5DZY  Davis, Perkins and Bishop Inc            Restaurants   
1  TitleKIW18                 Bender PLC LLC  Construction Services   
2  TitleFXSDN          Carter-Henry and Sons      Hospitals/Clinics   
3  TitlePSK4Y                 Garcia Ltd Ltd            Real Estate   
4  Title904GV               Lee and Sons PLC     Financial Services   

   Deal_value  Weighted_amount Date_of_creation      Pitch  \
0    320506.0     2.067264e+06       2020-03-29  Product_2   
1     39488.0     2.408768e+05       2019-07-10  Product_2   
2    359392.0     2.407926e+06       2019-07-27  Product_1   
3     76774.0     4.683214e+05       2021-01-30  Product_2   
4    483896.0     1.569884e+06       2019-05-22  Product_2   

             Contact_no             Lead_revenue Fund_category Geography  \
0          607.447.7883         50 - 100 Million    Category 2       USA   
1          892-938-9493  500 Million - 1 Billion    Category 4     India   
2          538.748.2271  500 Million - 1 Billion    Category 4       USA   
3   (692)052-1389x75188  500 Million - 1 Billion    Category 3       USA   
4  001-878-814-6134x015         50 - 100 Million    Category 3     India   

                      Location         POC_name               Designation  \
0           Killeen-Temple, TX  Charlene Werner  Executive Vice President   
1                       Ratlam            rakhi    Chairman/CEO/President   
2  Albany-Schenectady-Troy, NY   Ariel Hamilton      SVP/General Counsel    
3   Mount Vernon-Anacortes, WA      Erin Wilson   CEO/Co-Founder/Chairman   
4                      Shimoga           kavita  Executive Vice President   

                  Lead_POC_email           Hiring_candidate_role  \
0       charlenewerner@davis.com            Community pharmacist   
1          terrylogan@bender.com          Recruitment consultant   
2  arielhamilton@carterhenry.com          Health service manager   
3          erinwilson@garcia.com  Therapist, speech and language   
4         mr.christopher@lee.com                   Media planner   

       Lead_source Level_of_meeting                 Last_lead_update  \
0          Website          Level 3                         No track   
1           Others          Level 1  Did not hear back after Level 1   
2  Marketing Event          Level 1                                ?   
3    Contact Email          Level 2  Did not hear back after Level 1   
4          Website          Level 2                       Up-to-date   

                Internal_POC                      Resource  Internal_rating  \
0           Davis,Sharrice A                            No                3   
1             Brown,Maxine A                            No                5   
2  Georgakopoulos,Vasilios T                            No                4   
3             Brown,Maxine A  We have all the requirements                1   
4              Thomas,Lori E                            No                4   

   Success_probability  
0                 73.6  
1                 58.9  
2                 68.8  
3                 64.5  
4                 62.4

In [ ]:
# Starting from first variable.
#Lead_name consist of all unique values.

In [199]:
data2.head()

Deal_title                         Lead_name                  Industry  \
0  TitleM5DZY     Davis, Perkins and Bishop Inc               Restaurants   
2  TitleFXSDN             Carter-Henry and Sons         Hospitals/Clinics   
3  TitlePSK4Y                    Garcia Ltd Ltd               Real Estate   
6  TitleOZQRY           Williamson LLC and Sons                     Banks   
7  TitleV05WV  Livingston, York and Adams Group  Architecture/Engineering   

   Deal_value  Weighted_amount Date_of_creation      Pitch  \
0  320,506.00     2,067,263.70       2020-03-29  Product_2   
2  359,392.00     2,407,926.40       2019-07-27  Product_1   
3   76,774.00       468,321.40       2021-01-30  Product_2   
6  384,356.00     2,709,709.80       2019-11-20  Product_2   
7  245,205.00     1,642,873.50       2020-12-22  Product_2   

            Contact_no             Lead_revenue Fund_category Geography  \
0         607.447.7883         50 - 100 Million    Category 2       USA   
2         538.748.2271  500 Million - 1 Billion    Category 4       USA   
3  (692)052-1389x75188  500 Million - 1 Billion    Category 3       USA   
6        (881)077-4692  500 Million - 1 Billion    Category 1       USA   
7         953.762.9149  500 Million - 1 Billion    Category 1       USA   

                      Location           POC_name  \
0           Killeen-Temple, TX    Charlene Werner   
2  Albany-Schenectady-Troy, NY     Ariel Hamilton   
3   Mount Vernon-Anacortes, WA        Erin Wilson   
6             Salisbury, MD-DE         Sara Dixon   
7             Jacksonville, FL  Christian Stanley   

                                         Designation  \
0                           Executive Vice President   
2                               SVP/General Counsel    
3                            CEO/Co-Founder/Chairman   
6                            CEO/Co-Founder/Chairman   
7  Vice President / GM (04-present) : VP Sales an...   

                    Lead_POC_email           Hiring_candidate_role  \
0         charlenewerner@davis.com            Community pharmacist   
2    arielhamilton@carterhenry.com          Health service manager   
3            erinwilson@garcia.com  Therapist, speech and language   
6         saradixon@williamson.com                    Cartographer   
7  christianstanley@livingston.com      Engineer, maintenance (IT)   

       Lead_source Level_of_meeting                 Last_lead_update  \
0          Website          Level 3                         No track   
2  Marketing Event          Level 1                                ?   
3    Contact Email          Level 2  Did not hear back after Level 1   
6    Contact Email          Level 3                More than 2 weeks   
7           Others          Level 2                      5 days back   

                Internal_POC                      Resource  Internal_rating  \
0           Davis,Sharrice A                           NaN                3   
2  Georgakopoulos,Vasilios T                            No                4   
3             Brown,Maxine A  We have all the requirements                1   
6             Booker,David L                           NaN                4   
7              Cashin,Marc C                            No                5   

   Success_probability  
0                73.60  
2                68.80  
3                64.50  
6                73.40  
7                74.10

In [200]:
data["Lead_name"]= lab_encod.fit_transform(data["Lead_name"])

In [211]:
#Count of unique values in Industry column
data["Industry"].unique().shape

(171,)

In [212]:
#lets label encode them
data["Industry"]= lab_encod.fit_transform(data["Industry"])

In [215]:
##Count of unique values in Pitch column
data["Pitch"].unique()

array(['Product_2', 'Product_1'], dtype=object)

In [216]:
data["Pitch"]= lab_encod.fit_transform(data["Pitch"])

In [217]:
data.head()

Deal_title  Lead_name  Industry  Deal_value  Weighted_amount  \
0  TitleM5DZY       1386       119  320,506.00     2,067,263.70   
1  TitleKIW18        411        31   39,488.00       240,876.80   
2  TitleFXSDN        940        56  359,392.00     2,407,926.40   
3  TitlePSK4Y       1943       114   76,774.00       468,321.40   
4  Title904GV       3477        46  483,896.00     1,569,883.67   

  Date_of_creation  Pitch            Contact_no             Lead_revenue  \
0       2020-03-29      1          607.447.7883         50 - 100 Million   
1       2019-07-10      1          892-938-9493  500 Million - 1 Billion   
2       2019-07-27      0          538.748.2271  500 Million - 1 Billion   
3       2021-01-30      1   (692)052-1389x75188  500 Million - 1 Billion   
4       2019-05-22      1  001-878-814-6134x015         50 - 100 Million   

  Fund_category Geography                     Location         POC_name  \
0    Category 2       USA           Killeen-Temple, TX  Charlene Werner   
1    Category 4     India                       Ratlam            rakhi   
2    Category 4       USA  Albany-Schenectady-Troy, NY   Ariel Hamilton   
3    Category 3       USA   Mount Vernon-Anacortes, WA      Erin Wilson   
4    Category 3     India                      Shimoga           kavita   

                Designation                 Lead_POC_email  \
0  Executive Vice President       charlenewerner@davis.com   
1    Chairman/CEO/President          terrylogan@bender.com   
2      SVP/General Counsel   arielhamilton@carterhenry.com   
3   CEO/Co-Founder/Chairman          erinwilson@garcia.com   
4  Executive Vice President         mr.christopher@lee.com   

            Hiring_candidate_role      Lead_source Level_of_meeting  \
0            Community pharmacist          Website          Level 3   
1          Recruitment consultant           Others          Level 1   
2          Health service manager  Marketing Event          Level 1   
3  Therapist, speech and language    Contact Email          Level 2   
4                   Media planner          Website          Level 2   

                  Last_lead_update               Internal_POC  \
0                         No track           Davis,Sharrice A   
1  Did not hear back after Level 1             Brown,Maxine A   
2                                ?  Georgakopoulos,Vasilios T   
3  Did not hear back after Level 1             Brown,Maxine A   
4                       Up-to-date              Thomas,Lori E   

                       Resource  Internal_rating  Success_probability  
0                            No                3                73.60  
1                            No                5                58.90  
2                            No                4                68.80  
3  We have all the requirements                1                64.50  
4                            No                4                62.40

In [221]:
#we will drop contact no
#Now lead revenue 
data["Lead_revenue"].unique()

(3,)

In [239]:
data= pd.get_dummies(data,columns=["Lead_revenue"], drop_first=True)

In [241]:
data.Fund_category.unique()

array(['Category 2', 'Category 4', 'Category 3', 'Category 1'],
      dtype=object)

In [242]:
#label encode Fund_category
data["Fund_category"]= lab_encod.fit_transform(data["Fund_category"])

In [246]:
#label encode Geography
data["Geography"]= lab_encod.fit_transform(data["Geography"])

In [250]:
#label encode Location
data.Location.unique().shape

(597,)

In [251]:
#label encode Location
data["Location"]= lab_encod.fit_transform(data["Location"])

In [254]:
data.POC_name.unique().shape
#i dont thik POC name is of much use. so we will drop it

(5261,)

In [257]:
#label encode Designation
data["Designation"]= lab_encod.fit_transform(data["Designation"])

In [259]:
#i dont thik Lead_POC_email is of much importance we will drop it and contact number, Poc Name

(7007,)

In [262]:
#how much unique values Hiring_candidate_role have 
data.Hiring_candidate_role.unique().shape

(639,)

In [263]:
# Hiring_candidate_role as it can be a good feature
#label encode Hiring_candidate_role
data["Hiring_candidate_role"]= lab_encod.fit_transform(data["Hiring_candidate_role"])

In [266]:
#how much unique values Lead_source have 
data.Lead_source.unique()

array(['Website', 'Others', 'Marketing Event', 'Contact Email'],
      dtype=object)

In [267]:
#label encode Lead_source
data["Lead_source"]= lab_encod.fit_transform(data["Lead_source"])

In [269]:
data.Level_of_meeting.unique()

array(['Level 3', 'Level 1', 'Level 2'], dtype=object)

In [270]:
#level of meating is an important topic to be considered. thus we will encode tem using get dummies to create feature.
data= pd.get_dummies(data,columns=["Level_of_meeting"], drop_first=True)

In [275]:
data.Last_lead_update.unique()

array(['No track', 'Did not hear back after Level 1', '?', 'Up-to-date',
       '2 days back', 'More than 2 weeks', '5 days back',
       'more than a month', 'Following up but lead not responding',
       'More than a week back', 'Pending'], dtype=object)

In [279]:
#first we will replace then ? with unknown 
data["Last_lead_update"] = data["Last_lead_update"].replace("?","Unknown")

In [281]:
#label encode Last_lead_update
data["Last_lead_update"]= lab_encod.fit_transform(data["Last_lead_update"])

In [284]:
#Internal_POC consist of 60 unique values this can be useful. Lets label encode them.
data.Internal_POC.unique().shape 

(60,)

In [285]:
#label encode Last_lead_update
data["Internal_POC"]= lab_encod.fit_transform(data["Internal_POC"])

In [288]:
#we have 6 unique values in the Resource column 
data.Resource.unique().shape

(6,)

In [289]:
#label encode Last_lead_update
data["Resource"]= lab_encod.fit_transform(data["Resource"])

In [ ]:
# now we have completed all the encoding work lets se the data frame

In [290]:
data.head()

Deal_title  Lead_name  Industry  Deal_value  Weighted_amount  \
0  TitleM5DZY       1386       119  320,506.00     2,067,263.70   
1  TitleKIW18        411        31   39,488.00       240,876.80   
2  TitleFXSDN        940        56  359,392.00     2,407,926.40   
3  TitlePSK4Y       1943       114   76,774.00       468,321.40   
4  Title904GV       3477        46  483,896.00     1,569,883.67   

  Date_of_creation  Pitch            Contact_no  Fund_category  Geography  \
0       2020-03-29      1          607.447.7883              1          1   
1       2019-07-10      1          892-938-9493              3          0   
2       2019-07-27      0          538.748.2271              3          1   
3       2021-01-30      1   (692)052-1389x75188              2          1   
4       2019-05-22      1  001-878-814-6134x015              2          0   

   Location         POC_name  Designation                 Lead_POC_email  \
0       292  Charlene Werner            7       charlenewerner@davis.com   
1       457            rakhi            4          terrylogan@bender.com   
2        15   Ariel Hamilton            8  arielhamilton@carterhenry.com   
3       372      Erin Wilson            2          erinwilson@garcia.com   
4       505           kavita            7         mr.christopher@lee.com   

   Hiring_candidate_role  Lead_source  Last_lead_update  Internal_POC  \
0                    126            3                 6            12   
1                    501            2                 2             5   
2                    286            1                 8            18   
3                    611            0                 2             5   
4                    369            3                 9            53   

   Resource  Internal_rating  Success_probability  \
0         2                3                73.60   
1         2                5                58.90   
2         2                4                68.80   
3         4                1                64.50   
4         2                4                62.40   

   Lead_revenue_50 - 100 Million  Lead_revenue_500 Million - 1 Billion  \
0                              1                                     0   
1                              0                                     1   
2                              0                                     1   
3                              0                                     1   
4                              1                                     0   

   Level_of_meeting_Level 2  Level_of_meeting_Level 3  
0                         0                         1  
1                         0                         0  
2                         0                         0  
3                         1                         0  
4                         1                         0

In [292]:
data.to_csv("Encoded.csv", index=False)

In [293]:
# now lets delete the columns
data.drop(columns=["Lead_name","Contact_no","Lead_POC_email"], inplace=True)

In [294]:
data.head()

Deal_title  Industry  Deal_value  Weighted_amount Date_of_creation  Pitch  \
0  TitleM5DZY       119  320,506.00     2,067,263.70       2020-03-29      1   
1  TitleKIW18        31   39,488.00       240,876.80       2019-07-10      1   
2  TitleFXSDN        56  359,392.00     2,407,926.40       2019-07-27      0   
3  TitlePSK4Y       114   76,774.00       468,321.40       2021-01-30      1   
4  Title904GV        46  483,896.00     1,569,883.67       2019-05-22      1   

   Fund_category  Geography  Location         POC_name  Designation  \
0              1          1       292  Charlene Werner            7   
1              3          0       457            rakhi            4   
2              3          1        15   Ariel Hamilton            8   
3              2          1       372      Erin Wilson            2   
4              2          0       505           kavita            7   

   Hiring_candidate_role  Lead_source  Last_lead_update  Internal_POC  \
0                    126            3                 6            12   
1                    501            2                 2             5   
2                    286            1                 8            18   
3                    611            0                 2             5   
4                    369            3                 9            53   

   Resource  Internal_rating  Success_probability  \
0         2                3                73.60   
1         2                5                58.90   
2         2                4                68.80   
3         4                1                64.50   
4         2                4                62.40   

   Lead_revenue_50 - 100 Million  Lead_revenue_500 Million - 1 Billion  \
0                              1                                     0   
1                              0                                     1   
2                              0                                     1   
3                              0                                     1   
4                              1                                     0   

   Level_of_meeting_Level 2  Level_of_meeting_Level 3  
0                         0                         1  
1                         0                         0  
2                         0                         0  
3                         1                         0  
4                         1                         0

In [295]:
#now we will drop records with Success_Probability with value over 100.
final_data=data[data["Success_probability"]<100]

In [297]:
final_data.head()

Deal_title  Industry  Deal_value  Weighted_amount Date_of_creation  Pitch  \
0  TitleM5DZY       119  320,506.00     2,067,263.70       2020-03-29      1   
1  TitleKIW18        31   39,488.00       240,876.80       2019-07-10      1   
2  TitleFXSDN        56  359,392.00     2,407,926.40       2019-07-27      0   
3  TitlePSK4Y       114   76,774.00       468,321.40       2021-01-30      1   
4  Title904GV        46  483,896.00     1,569,883.67       2019-05-22      1   

   Fund_category  Geography  Location         POC_name  Designation  \
0              1          1       292  Charlene Werner            7   
1              3          0       457            rakhi            4   
2              3          1        15   Ariel Hamilton            8   
3              2          1       372      Erin Wilson            2   
4              2          0       505           kavita            7   

   Hiring_candidate_role  Lead_source  Last_lead_update  Internal_POC  \
0                    126            3                 6            12   
1                    501            2                 2             5   
2                    286            1                 8            18   
3                    611            0                 2             5   
4                    369            3                 9            53   

   Resource  Internal_rating  Success_probability  \
0         2                3                73.60   
1         2                5                58.90   
2         2                4                68.80   
3         4                1                64.50   
4         2                4                62.40   

   Lead_revenue_50 - 100 Million  Lead_revenue_500 Million - 1 Billion  \
0                              1                                     0   
1                              0                                     1   
2                              0                                     1   
3                              0                                     1   
4                              1                                     0   

   Level_of_meeting_Level 2  Level_of_meeting_Level 3  
0                         0                         1  
1                         0                         0  
2                         0                         0  
3                         1                         0  
4                         1                         0

In [300]:
final_data.POC_name.unique().shape

(4914,)

In [302]:
#lets also encode POC_name 
final_data["POC_name"]= lab_encod.fit_transform(final_data["POC_name"])

<ipython-input-302-4b0c868d69ce>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  final_data["POC_name"]= lab_encod.fit_transform(final_data["POC_name"])


In [306]:
final_data.head()

Deal_title  Industry  Deal_value  Weighted_amount  Date_of_creation  Pitch  \
0  TitleM5DZY       119  320,506.00     2,067,263.70               453      1   
1  TitleKIW18        31   39,488.00       240,876.80               190      1   
2  TitleFXSDN        56  359,392.00     2,407,926.40               207      0   
3  TitlePSK4Y       114   76,774.00       468,321.40               760      1   
4  Title904GV        46  483,896.00     1,569,883.67               141      1   

   Fund_category  Geography  Location  POC_name  Designation  \
0              1          1       292       517            7   
1              3          0       457      4219            4   
2              3          1        15       240            8   
3              2          1       372      1069            2   
4              2          0       505      3675            7   

   Hiring_candidate_role  Lead_source  Last_lead_update  Internal_POC  \
0                    126            3                 6            12   
1                    501            2                 2             5   
2                    286            1                 8            18   
3                    611            0                 2             5   
4                    369            3                 9            53   

   Resource  Internal_rating  Success_probability  \
0         2                3                73.60   
1         2                5                58.90   
2         2                4                68.80   
3         4                1                64.50   
4         2                4                62.40   

   Lead_revenue_50 - 100 Million  Lead_revenue_500 Million - 1 Billion  \
0                              1                                     0   
1                              0                                     1   
2                              0                                     1   
3                              0                                     1   
4                              1                                     0   

   Level_of_meeting_Level 2  Level_of_meeting_Level 3  
0                         0                         1  
1                         0                         0  
2                         0                         0  
3                         1                         0  
4                         1                         0

In [305]:
#data.Date_of_creation.unique().shape
#lets also encode the Date_of_creation
final_data["Date_of_creation"]= lab_encod.fit_transform(final_data["Date_of_creation"])

<ipython-input-305-87621d22c2b5>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  final_data["Date_of_creation"]= lab_encod.fit_transform(final_data["Date_of_creation"])


In [333]:
final_data.head()

Deal_title  Industry  Deal_value  Weighted_amount  Date_of_creation  Pitch  \
0  TitleM5DZY       119  320,506.00     2,067,263.70               453      1   
1  TitleKIW18        31   39,488.00       240,876.80               190      1   
2  TitleFXSDN        56  359,392.00     2,407,926.40               207      0   
3  TitlePSK4Y       114   76,774.00       468,321.40               760      1   
4  Title904GV        46  483,896.00     1,569,883.67               141      1   

   Fund_category  Geography  Location  POC_name  Designation  \
0              1          1       292       517            7   
1              3          0       457      4219            4   
2              3          1        15       240            8   
3              2          1       372      1069            2   
4              2          0       505      3675            7   

   Hiring_candidate_role  Lead_source  Last_lead_update  Internal_POC  \
0                    126            3                 6            12   
1                    501            2                 2             5   
2                    286            1                 8            18   
3                    611            0                 2             5   
4                    369            3                 9            53   

   Resource  Internal_rating  Success_probability  \
0         2                3                73.60   
1         2                5                58.90   
2         2                4                68.80   
3         4                1                64.50   
4         2                4                62.40   

   Lead_revenue_50 - 100 Million  Lead_revenue_500 Million - 1 Billion  \
0                              1                                     0   
1                              0                                     1   
2                              0                                     1   
3                              0                                     1   
4                              1                                     0   

   Level_of_meeting_Level 2  Level_of_meeting_Level 3  
0                         0                         1  
1                         0                         0  
2                         0                         0  
3                         1                         0  
4                         1                         0

In [334]:
#set Deal_title as index
final_data.set_index("Deal_title", inplace=True)

Industry  Deal_value  Weighted_amount  Date_of_creation  Pitch  \
Deal_title                                                                   
TitleM5DZY       119  320,506.00     2,067,263.70               453      1   
TitleKIW18        31   39,488.00       240,876.80               190      1   
TitleFXSDN        56  359,392.00     2,407,926.40               207      0   
TitlePSK4Y       114   76,774.00       468,321.40               760      1   
Title904GV        46  483,896.00     1,569,883.67               141      1   

            Fund_category  Geography  Location  POC_name  Designation  \
Deal_title                                                              
TitleM5DZY              1          1       292       517            7   
TitleKIW18              3          0       457      4219            4   
TitleFXSDN              3          1        15       240            8   
TitlePSK4Y              2          1       372      1069            2   
Title904GV              2          0       505      3675            7   

            Hiring_candidate_role  Lead_source  Last_lead_update  \
Deal_title                                                         
TitleM5DZY                    126            3                 6   
TitleKIW18                    501            2                 2   
TitleFXSDN                    286            1                 8   
TitlePSK4Y                    611            0                 2   
Title904GV                    369            3                 9   

            Internal_POC  Resource  Internal_rating  Success_probability  \
Deal_title                                                                 
TitleM5DZY            12         2                3                73.60   
TitleKIW18             5         2                5                58.90   
TitleFXSDN            18         2                4                68.80   
TitlePSK4Y             5         4                1                64.50   
Title904GV            53         2                4                62.40   

            Lead_revenue_50 - 100 Million  \
Deal_title                                  
TitleM5DZY                              1   
TitleKIW18                              0   
TitleFXSDN                              0   
TitlePSK4Y                              0   
Title904GV                              1   

            Lead_revenue_500 Million - 1 Billion  Level_of_meeting_Level 2  \
Deal_title                                                                   
TitleM5DZY                                     0                         0   
TitleKIW18                                     1                         0   
TitleFXSDN                                     1                         0   
TitlePSK4Y                                     1                         1   
Title904GV                                     0                         1   

            Level_of_meeting_Level 3  
Deal_title                            
TitleM5DZY                         1  
TitleKIW18                         0  
TitleFXSDN                         0  
TitlePSK4Y                         0  
Title904GV                         0

In [336]:
final_data.columns

Index(['Industry', 'Deal_value', 'Weighted_amount', 'Date_of_creation',
       'Pitch', 'Fund_category', 'Geography', 'Location', 'POC_name',
       'Designation', 'Hiring_candidate_role', 'Lead_source',
       'Last_lead_update', 'Internal_POC', 'Resource', 'Internal_rating',
       'Success_probability', 'Lead_revenue_50 - 100 Million',
       'Lead_revenue_500 Million - 1 Billion', 'Level_of_meeting_Level 2',
       'Level_of_meeting_Level 3'],
      dtype='object')

In [350]:
fdata=final_data.reindex(['Industry', 'Deal_value', 'Weighted_amount','Date_of_creation',
       'Pitch', 'Fund_category', 'Geography', 'Location', 'POC_name',
       'Designation', 'Hiring_candidate_role', 'Lead_source',
       'Last_lead_update', 'Internal_POC', 'Resource', 'Internal_rating','Lead_revenue_50 - 100 Million',
       'Lead_revenue_500 Million - 1 Billion', 'Level_of_meeting_Level 2',
       'Level_of_meeting_Level 3','Success_probability'], axis=1)

In [351]:
fdata.head()

Industry  Deal_value  Weighted_amount  Date_of_creation  Pitch  \
Deal_title                                                                   
TitleM5DZY       119  320,506.00     2,067,263.70               453      1   
TitleKIW18        31   39,488.00       240,876.80               190      1   
TitleFXSDN        56  359,392.00     2,407,926.40               207      0   
TitlePSK4Y       114   76,774.00       468,321.40               760      1   
Title904GV        46  483,896.00     1,569,883.67               141      1   

            Fund_category  Geography  Location  POC_name  Designation  \
Deal_title                                                              
TitleM5DZY              1          1       292       517            7   
TitleKIW18              3          0       457      4219            4   
TitleFXSDN              3          1        15       240            8   
TitlePSK4Y              2          1       372      1069            2   
Title904GV              2          0       505      3675            7   

            Hiring_candidate_role  Lead_source  Last_lead_update  \
Deal_title                                                         
TitleM5DZY                    126            3                 6   
TitleKIW18                    501            2                 2   
TitleFXSDN                    286            1                 8   
TitlePSK4Y                    611            0                 2   
Title904GV                    369            3                 9   

            Internal_POC  Resource  Internal_rating  \
Deal_title                                            
TitleM5DZY            12         2                3   
TitleKIW18             5         2                5   
TitleFXSDN            18         2                4   
TitlePSK4Y             5         4                1   
Title904GV            53         2                4   

            Lead_revenue_50 - 100 Million  \
Deal_title                                  
TitleM5DZY                              1   
TitleKIW18                              0   
TitleFXSDN                              0   
TitlePSK4Y                              0   
Title904GV                              1   

            Lead_revenue_500 Million - 1 Billion  Level_of_meeting_Level 2  \
Deal_title                                                                   
TitleM5DZY                                     0                         0   
TitleKIW18                                     1                         0   
TitleFXSDN                                     1                         0   
TitlePSK4Y                                     1                         1   
Title904GV                                     0                         1   

            Level_of_meeting_Level 3  Success_probability  
Deal_title                                                 
TitleM5DZY                         1                73.60  
TitleKIW18                         0                58.90  
TitleFXSDN                         0                68.80  
TitlePSK4Y                         0                64.50  
Title904GV                         0                62.40

In [352]:
X=fdata.copy()
X.drop(columns=["Success_probability"], inplace=True)
X.head()

Industry  Deal_value  Weighted_amount  Date_of_creation  Pitch  \
Deal_title                                                                   
TitleM5DZY       119  320,506.00     2,067,263.70               453      1   
TitleKIW18        31   39,488.00       240,876.80               190      1   
TitleFXSDN        56  359,392.00     2,407,926.40               207      0   
TitlePSK4Y       114   76,774.00       468,321.40               760      1   
Title904GV        46  483,896.00     1,569,883.67               141      1   

            Fund_category  Geography  Location  POC_name  Designation  \
Deal_title                                                              
TitleM5DZY              1          1       292       517            7   
TitleKIW18              3          0       457      4219            4   
TitleFXSDN              3          1        15       240            8   
TitlePSK4Y              2          1       372      1069            2   
Title904GV              2          0       505      3675            7   

            Hiring_candidate_role  Lead_source  Last_lead_update  \
Deal_title                                                         
TitleM5DZY                    126            3                 6   
TitleKIW18                    501            2                 2   
TitleFXSDN                    286            1                 8   
TitlePSK4Y                    611            0                 2   
Title904GV                    369            3                 9   

            Internal_POC  Resource  Internal_rating  \
Deal_title                                            
TitleM5DZY            12         2                3   
TitleKIW18             5         2                5   
TitleFXSDN            18         2                4   
TitlePSK4Y             5         4                1   
Title904GV            53         2                4   

            Lead_revenue_50 - 100 Million  \
Deal_title                                  
TitleM5DZY                              1   
TitleKIW18                              0   
TitleFXSDN                              0   
TitlePSK4Y                              0   
Title904GV                              1   

            Lead_revenue_500 Million - 1 Billion  Level_of_meeting_Level 2  \
Deal_title                                                                   
TitleM5DZY                                     0                         0   
TitleKIW18                                     1                         0   
TitleFXSDN                                     1                         0   
TitlePSK4Y                                     1                         1   
Title904GV                                     0                         1   

            Level_of_meeting_Level 3  
Deal_title                            
TitleM5DZY                         1  
TitleKIW18                         0  
TitleFXSDN                         0  
TitlePSK4Y                         0  
Title904GV                         0

In [353]:
fdata.head()

Industry  Deal_value  Weighted_amount  Date_of_creation  Pitch  \
Deal_title                                                                   
TitleM5DZY       119  320,506.00     2,067,263.70               453      1   
TitleKIW18        31   39,488.00       240,876.80               190      1   
TitleFXSDN        56  359,392.00     2,407,926.40               207      0   
TitlePSK4Y       114   76,774.00       468,321.40               760      1   
Title904GV        46  483,896.00     1,569,883.67               141      1   

            Fund_category  Geography  Location  POC_name  Designation  \
Deal_title                                                              
TitleM5DZY              1          1       292       517            7   
TitleKIW18              3          0       457      4219            4   
TitleFXSDN              3          1        15       240            8   
TitlePSK4Y              2          1       372      1069            2   
Title904GV              2          0       505      3675            7   

            Hiring_candidate_role  Lead_source  Last_lead_update  \
Deal_title                                                         
TitleM5DZY                    126            3                 6   
TitleKIW18                    501            2                 2   
TitleFXSDN                    286            1                 8   
TitlePSK4Y                    611            0                 2   
Title904GV                    369            3                 9   

            Internal_POC  Resource  Internal_rating  \
Deal_title                                            
TitleM5DZY            12         2                3   
TitleKIW18             5         2                5   
TitleFXSDN            18         2                4   
TitlePSK4Y             5         4                1   
Title904GV            53         2                4   

            Lead_revenue_50 - 100 Million  \
Deal_title                                  
TitleM5DZY                              1   
TitleKIW18                              0   
TitleFXSDN                              0   
TitlePSK4Y                              0   
Title904GV                              1   

            Lead_revenue_500 Million - 1 Billion  Level_of_meeting_Level 2  \
Deal_title                                                                   
TitleM5DZY                                     0                         0   
TitleKIW18                                     1                         0   
TitleFXSDN                                     1                         0   
TitlePSK4Y                                     1                         1   
Title904GV                                     0                         1   

            Level_of_meeting_Level 3  Success_probability  
Deal_title                                                 
TitleM5DZY                         1                73.60  
TitleKIW18                         0                58.90  
TitleFXSDN                         0                68.80  
TitlePSK4Y                         0                64.50  
Title904GV                         0                62.40

In [354]:
Y=fdata
Y.drop(columns=['Industry', 'Deal_value', 'Weighted_amount','Date_of_creation',
       'Pitch', 'Fund_category', 'Geography', 'Location', 'POC_name',
       'Designation', 'Hiring_candidate_role', 'Lead_source',
       'Last_lead_update', 'Internal_POC', 'Resource', 'Internal_rating','Lead_revenue_50 - 100 Million',
       'Lead_revenue_500 Million - 1 Billion', 'Level_of_meeting_Level 2',
       'Level_of_meeting_Level 3'], inplace=True)
Y.head()

Success_probability
Deal_title                     
TitleM5DZY                73.60
TitleKIW18                58.90
TitleFXSDN                68.80
TitlePSK4Y                64.50
Title904GV                62.40

In [408]:
from sklearn.model_selection import train_test_split
from sklearn.model_selection import KFold
from sklearn.model_selection import cross_val_score
X_train, X_test, y_train, y_test = train_test_split(X, Y, train_size=0.70, random_state=2)

In [409]:
import xgboost as xgb

In [410]:
model = xgb.XGBRegressor(n_estimators=150,eval_metric='mae',max_depth=6)
model.fit(X_train, y_train, early_stopping_rounds=35, eval_set=[(X_test, y_test)], verbose=1)

[0]	validation_0-mae:43.37427
[1]	validation_0-mae:31.78514
[2]	validation_0-mae:24.35595
[3]	validation_0-mae:19.14029
[4]	validation_0-mae:15.51603
[5]	validation_0-mae:12.93877
[6]	validation_0-mae:11.16742
[7]	validation_0-mae:9.94884
[8]	validation_0-mae:9.03790
[9]	validation_0-mae:8.42927
[10]	validation_0-mae:8.00132
[11]	validation_0-mae:7.73434
[12]	validation_0-mae:7.54235
[13]	validation_0-mae:7.40711
[14]	validation_0-mae:7.32424
[15]	validation_0-mae:7.27113
[16]	validation_0-mae:7.24728
[17]	validation_0-mae:7.24329
[18]	validation_0-mae:7.25392
[19]	validation_0-mae:7.25525
[20]	validation_0-mae:7.29057
[21]	validation_0-mae:7.29308
[22]	validation_0-mae:7.30279
[23]	validation_0-mae:7.30006
[24]	validation_0-mae:7.29427
[25]	validation_0-mae:7.35021
[26]	validation_0-mae:7.36814
[27]	validation_0-mae:7.37947
[28]	validation_0-mae:7.37974
[29]	validation_0-mae:7.39530
[30]	validation_0-mae:7.40354
[31]	validation_0-mae:7.41031
[32]	validation_0-mae:7.41141
[33]	validati

XGBRegressor(base_score=0.5, booster='gbtree', colsample_bylevel=1,
             colsample_bynode=1, colsample_bytree=1, eval_metric='mae', gamma=0,
             gpu_id=-1, importance_type='gain', interaction_constraints='',
             learning_rate=0.300000012, max_delta_step=0, max_depth=6,
             min_child_weight=1, missing=nan, monotone_constraints='()',
             n_estimators=150, n_jobs=4, num_parallel_tree=1, random_state=0,
             reg_alpha=0, reg_lambda=1, scale_pos_weight=1, subsample=1,
             tree_method='exact', validate_parameters=1, verbosity=None)

Refer para
XGBRegressor(base_score=0.5, booster='gbtree', colsample_bylevel=1,
             colsample_bynode=1, colsample_bytree=1, eval_metric='mae', gamma=0,
             gpu_id=-1, importance_type='gain', interaction_constraints='',
             learning_rate=0.300000012, max_delta_step=0, max_depth=6,
             min_child_weight=1, missing=nan, monotone_constraints='()',
             n_estimators=100, n_jobs=4, num_parallel_tree=1, random_state=0,
             reg_alpha=0, reg_lambda=1, scale_pos_weight=1, subsample=1,
             tree_method='exact', validate_parameters=1, verbosity=None)

In [411]:
ypred=model.predict(X_test)

In [412]:
from sklearn import metrics
score=max(0,100-np.sqrt(metrics.mean_squared_error(y_test,ypred)))
print(score)

86.873697030073


In [413]:
X_train.head()

Industry  Deal_value  Weighted_amount  Date_of_creation  Pitch  \
Deal_title                                                                   
TitleU5I69        62  226,834.00     1,485,762.70               169      0   
TitleC1JWL         0  202,313.00     1,183,531.05               694      0   
TitleWII3C        66  115,699.00       682,624.10               220      0   
Title6AV9M         2   98,010.00       656,667.00                77      0   
TitleOVF7D        31  461,058.00     3,042,982.80               558      1   

            Fund_category  Geography  Location  POC_name  Designation  \
Deal_title                                                              
TitleU5I69              2          0        18      3998            6   
TitleC1JWL              2          0       113      3205            3   
TitleWII3C              2          0       377      3702            0   
Title6AV9M              3          0       288      3246            9   
TitleOVF7D              2          1       249        96            7   

            Hiring_candidate_role  Lead_source  Last_lead_update  \
Deal_title                                                         
TitleU5I69                    254            1                 1   
TitleC1JWL                    523            0                 8   
TitleWII3C                    501            0                 9   
Title6AV9M                     65            3                 0   
TitleOVF7D                    349            1                 3   

            Internal_POC  Resource  Internal_rating  \
Deal_title                                            
TitleU5I69            55         0                4   
TitleC1JWL            27         4                3   
TitleWII3C            54         4                4   
Title6AV9M            22         4                2   
TitleOVF7D            28         1                5   

            Lead_revenue_50 - 100 Million  \
Deal_title                                  
TitleU5I69                              0   
TitleC1JWL                              1   
TitleWII3C                              0   
Title6AV9M                              0   
TitleOVF7D                              0   

            Lead_revenue_500 Million - 1 Billion  Level_of_meeting_Level 2  \
Deal_title                                                                   
TitleU5I69                                     0                         0   
TitleC1JWL                                     0                         1   
TitleWII3C                                     1                         0   
Title6AV9M                                     0                         0   
TitleOVF7D                                     0                         1   

            Level_of_meeting_Level 3  
Deal_title                            
TitleU5I69                         1  
TitleC1JWL                         0  
TitleWII3C                         0  
Title6AV9M                         1  
TitleOVF7D                         0



Now test The data

In [425]:
tdata=pd.read_csv("final_test1.csv")
tdata.head()

Deal_title  Industry  Deal_value  Weighted_amount  Date_of_creation  Pitch  \
0  TitleAD16O        55  200,988.00     1,556,916.67               427      0   
1  TitleOW6CR        30  409,961.00     2,541,758.20               698      0   
2  TitleVVJQ5        13  434,433.00     3,041,031.00               519      0   
3  TitleUS8NA        69  218,952.00     1,521,716.40               385      1   
4  Title5VGWW        95  392,835.00     2,455,218.75               610      0   

   Fund_category  Geography  Location  POC_name  Designation  \
0              3          0        68      1666            4   
1              2          1       122       220            2   
2              0          1       409        44            5   
3              0          0        48      1583            1   
4              2          1       522       901            2   

   Hiring_candidate_role  Lead_source  Last_lead_update  Internal_POC  \
0                    151            1                10            38   
1                    286            1                 9            51   
2                    307            1                 3            19   
3                    115            0                10            42   
4                    615            2                 5            42   

   Resource  Internal_rating  Lead_revenue_50 - 100 Million  \
0         2            -1.00                              0   
1         5             1.00                              0   
2         1             5.00                              0   
3         2             5.00                              0   
4         1             2.00                              0   

   Lead_revenue_500 Million - 1 Billion  Level_of_meeting_Level 2  \
0                                     0                         0   
1                                     0                         1   
2                                     0                         1   
3                                     0                         0   
4                                     1                         0   

   Level_of_meeting_Level 3  
0                         0  
1                         0  
2                         0  
3                         1  
4                         1

In [426]:
tdata.set_index("Deal_title", inplace=True)
tdata.head()

Industry  Deal_value  Weighted_amount  Date_of_creation  Pitch  \
Deal_title                                                                   
TitleAD16O        55  200,988.00     1,556,916.67               427      0   
TitleOW6CR        30  409,961.00     2,541,758.20               698      0   
TitleVVJQ5        13  434,433.00     3,041,031.00               519      0   
TitleUS8NA        69  218,952.00     1,521,716.40               385      1   
Title5VGWW        95  392,835.00     2,455,218.75               610      0   

            Fund_category  Geography  Location  POC_name  Designation  \
Deal_title                                                              
TitleAD16O              3          0        68      1666            4   
TitleOW6CR              2          1       122       220            2   
TitleVVJQ5              0          1       409        44            5   
TitleUS8NA              0          0        48      1583            1   
Title5VGWW              2          1       522       901            2   

            Hiring_candidate_role  Lead_source  Last_lead_update  \
Deal_title                                                         
TitleAD16O                    151            1                10   
TitleOW6CR                    286            1                 9   
TitleVVJQ5                    307            1                 3   
TitleUS8NA                    115            0                10   
Title5VGWW                    615            2                 5   

            Internal_POC  Resource  Internal_rating  \
Deal_title                                            
TitleAD16O            38         2            -1.00   
TitleOW6CR            51         5             1.00   
TitleVVJQ5            19         1             5.00   
TitleUS8NA            42         2             5.00   
Title5VGWW            42         1             2.00   

            Lead_revenue_50 - 100 Million  \
Deal_title                                  
TitleAD16O                              0   
TitleOW6CR                              0   
TitleVVJQ5                              0   
TitleUS8NA                              0   
Title5VGWW                              0   

            Lead_revenue_500 Million - 1 Billion  Level_of_meeting_Level 2  \
Deal_title                                                                   
TitleAD16O                                     0                         0   
TitleOW6CR                                     0                         1   
TitleVVJQ5                                     0                         1   
TitleUS8NA                                     0                         0   
Title5VGWW                                     1                         0   

            Level_of_meeting_Level 3  
Deal_title                            
TitleAD16O                         0  
TitleOW6CR                         0  
TitleVVJQ5                         0  
TitleUS8NA                         1  
Title5VGWW                         1

In [427]:
test_result=model.predict(tdata)

In [428]:
tdata["Success_probability"]=test_result

In [429]:
tdata.head()

Industry  Deal_value  Weighted_amount  Date_of_creation  Pitch  \
Deal_title                                                                   
TitleAD16O        55  200,988.00     1,556,916.67               427      0   
TitleOW6CR        30  409,961.00     2,541,758.20               698      0   
TitleVVJQ5        13  434,433.00     3,041,031.00               519      0   
TitleUS8NA        69  218,952.00     1,521,716.40               385      1   
Title5VGWW        95  392,835.00     2,455,218.75               610      0   

            Fund_category  Geography  Location  POC_name  Designation  \
Deal_title                                                              
TitleAD16O              3          0        68      1666            4   
TitleOW6CR              2          1       122       220            2   
TitleVVJQ5              0          1       409        44            5   
TitleUS8NA              0          0        48      1583            1   
Title5VGWW              2          1       522       901            2   

            Hiring_candidate_role  Lead_source  Last_lead_update  \
Deal_title                                                         
TitleAD16O                    151            1                10   
TitleOW6CR                    286            1                 9   
TitleVVJQ5                    307            1                 3   
TitleUS8NA                    115            0                10   
Title5VGWW                    615            2                 5   

            Internal_POC  Resource  Internal_rating  \
Deal_title                                            
TitleAD16O            38         2            -1.00   
TitleOW6CR            51         5             1.00   
TitleVVJQ5            19         1             5.00   
TitleUS8NA            42         2             5.00   
Title5VGWW            42         1             2.00   

            Lead_revenue_50 - 100 Million  \
Deal_title                                  
TitleAD16O                              0   
TitleOW6CR                              0   
TitleVVJQ5                              0   
TitleUS8NA                              0   
Title5VGWW                              0   

            Lead_revenue_500 Million - 1 Billion  Level_of_meeting_Level 2  \
Deal_title                                                                   
TitleAD16O                                     0                         0   
TitleOW6CR                                     0                         1   
TitleVVJQ5                                     0                         1   
TitleUS8NA                                     0                         0   
Title5VGWW                                     1                         0   

            Level_of_meeting_Level 3  Success_probability  
Deal_title                                                 
TitleAD16O                         0                59.99  
TitleOW6CR                         0                56.86  
TitleVVJQ5                         0                73.86  
TitleUS8NA                         1                73.36  
Title5VGWW                         1                65.45

In [430]:
#create a submission file
tdata.columns

Index(['Industry', 'Deal_value', 'Weighted_amount', 'Date_of_creation',
       'Pitch', 'Fund_category', 'Geography', 'Location', 'POC_name',
       'Designation', 'Hiring_candidate_role', 'Lead_source',
       'Last_lead_update', 'Internal_POC', 'Resource', 'Internal_rating',
       'Lead_revenue_50 - 100 Million', 'Lead_revenue_500 Million - 1 Billion',
       'Level_of_meeting_Level 2', 'Level_of_meeting_Level 3',
       'Success_probability'],
      dtype='object')

In [431]:
tdata.drop(columns=['Industry', 'Deal_value', 'Weighted_amount', 'Date_of_creation',
       'Pitch', 'Fund_category', 'Geography', 'Location', 'POC_name',
       'Designation', 'Hiring_candidate_role', 'Lead_source',
       'Last_lead_update', 'Internal_POC', 'Resource', 'Internal_rating',
       'Lead_revenue_50 - 100 Million', 'Lead_revenue_500 Million - 1 Billion',
       'Level_of_meeting_Level 2', 'Level_of_meeting_Level 3'], inplace=True)

In [432]:
tdata.to_csv("Submission_file1.csv", index=True)